### PR TITLE
Invite and remove the participant

### PR DESCRIPTION
**Pull request by jiahx**

### CODE REVIEW

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant.sln](https://github.com/jiahx/skype-docs/blob/master/Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant.sln) | :white_check_mark:Succeeded | |
[Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/App.config](https://github.com/jiahx/skype-docs/blob/master/Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/App.config) | :white_check_mark:Succeeded | |
[Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/InviteAndRemoveParticipant.csproj](https://github.com/jiahx/skype-docs/blob/master/Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/InviteAndRemoveParticipant.csproj) | :white_check_mark:Succeeded | |
[Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Program.cs](https://github.com/jiahx/skype-docs/blob/master/Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Program.cs) | :white_check_mark:Succeeded | |
[Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Properties/AssemblyInfo.cs](https://github.com/jiahx/skype-docs/blob/master/Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Properties/AssemblyInfo.cs) | :white_check_mark:Succeeded | |
[Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/packages.config](https://github.com/jiahx/skype-docs/blob/master/Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/packages.config) | :white_check_mark:Succeeded | |
[Skype/Trusted-Application-API/samples/QuickStartSamples/TrouterBasedEventChannel/Microsoft.Practices.Unity.dll](https://github.com/jiahx/skype-docs/blob/master/Skype/Trusted-Application-API/samples/QuickStartSamples/TrouterBasedEventChannel/Microsoft.Practices.Unity.dll) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C3%5C20%5C5315e434-bbfc-97f8-0e56-7036cb0b5e04%5CPullRequest%5C201703202018436147-227%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Program.cs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        private void Conversation_HandleParticipantChange(object sender, ParticipantChangeEventArgs eventArgs)
+        {
+            if (eventArgs.AddedParticipants?.Count > 0)
+            {
+                foreach (var participant in eventArgs.AddedParticipants)
+                {
+                    WriteToConsoleInColor(participant.Name + " has joined the meeting.");
+
+                    var participantUri = ConfigurationManager.AppSettings["ParticipantUri"];
+                    if (participantUri.Equals(participant.Uri))
```
Comment (sankum-msft) [inline Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Program.cs]: Can you please move this logic to RunAsync() method? See my comment above on how you can do that.

File: Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Reflection;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following 
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("TrustedJoinMeeting")]
```
Comment (sankum-msft) [inline Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Properties/AssemblyInfo.cs]: Please change the title

File: Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+[assembly: AssemblyDescription("")]
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("")]
+[assembly: AssemblyProduct("TrustedJoinMeeting")]
```
Comment (sankum-msft) [inline Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Properties/AssemblyInfo.cs]: Update this also

File: Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/InviteAndRemoveParticipant.csproj [hunk 4]
```diff
@@ -0,0 +1,147 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <ProjectGuid>{845BD88C-5FA8-43D7-B21F-9A57550DF727}</ProjectGuid>
+    <OutputType>Exe</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>TrustedJoinMeeting</RootNamespace>
```
Comment (sankum-msft) [inline Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/InviteAndRemoveParticipant.csproj]: Update this and the next line

File: Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Program.cs [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+            var invitation = await adhocMeeting.JoinAdhocMeeting(loggingContext, null).ConfigureAwait(false);
+
+            // Wait for the join to complete
+            await invitation.WaitForInviteCompleteAsync().ConfigureAwait(false);
+
+            invitation.RelatedConversation.HandleParticipantChange += Conversation_HandleParticipantChange;
+
+            // invite the participant to join the meeting
+            WriteToConsoleInColor("Invite " + participantUri+" to join the meeting");
+            await invitation.RelatedConversation.AddParticipantAsync(participantUri, loggingContext).ConfigureAwait(false);
```
Comment (sankum-msft) [inline Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Program.cs]: This starts an IParticipantInvitation. You can use WaitForInviteCompleteAsync() on it to wait for the participant to accept it and then eject the user.

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant.sln](https://github.com/jiahx/skype-docs/blob/master/Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant.sln) | :white_check_mark:Succeeded | |
[Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/App.config](https://github.com/jiahx/skype-docs/blob/master/Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/App.config) | :white_check_mark:Succeeded | |
[Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/InviteAndRemoveParticipant.csproj](https://github.com/jiahx/skype-docs/blob/master/Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/InviteAndRemoveParticipant.csproj) | :white_check_mark:Succeeded | |
[Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Program.cs](https://github.com/jiahx/skype-docs/blob/master/Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Program.cs) | :white_check_mark:Succeeded | |
[Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Properties/AssemblyInfo.cs](https://github.com/jiahx/skype-docs/blob/master/Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Properties/AssemblyInfo.cs) | :white_check_mark:Succeeded | |
[Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/packages.config](https://github.com/jiahx/skype-docs/blob/master/Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/packages.config) | :white_check_mark:Succeeded | |
[Skype/Trusted-Application-API/samples/QuickStartSamples/TrouterBasedEventChannel/Microsoft.Practices.Unity.dll](https://github.com/jiahx/skype-docs/blob/master/Skype/Trusted-Application-API/samples/QuickStartSamples/TrouterBasedEventChannel/Microsoft.Practices.Unity.dll) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C3%5C20%5C5315e434-bbfc-97f8-0e56-7036cb0b5e04%5CPullRequest%5C201703202208437610-227%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**